File: src/api/feeAndServices.ts
import { invoke } from "@tauri-apps/api/core";
import { TFeeAndServices } from "../types";
export const getFeeAndServicesApi = async () => {
  try {
    const result = await invoke<TFeeAndServices>("get_fee_and_services");
    return result;
  } catch (error) {
    console.log(error);
  }
};
export const addFeeAndServicesApi = async (data: TFeeAndServices) => {
  try {
    const res = await invoke<TFeeAndServices>("add_fee_and_services", {
      data,
    });
    return res;
  } catch (error) {
    console.error("Error adding fee and services:", error);
  }
};

export const updateFeeAndServicesApi = async (data: TFeeAndServices) => {
  try {
    const res = await invoke("update_fee_and_services", { data });

    return res;
  } catch (error) {
    console.error("Error updating fee and services:", error);
  }
};

File: src-tauri/src/handlers/fee_and_services.rs
use crate::datastore::fee_and_services;
use crate::types::FeeAndServices;

#[tauri::command]
pub fn add_fee_and_services(data: FeeAndServices, window: tauri::Window) -> Result<String, String> {
    fee_and_services::add_fee_and_services_db(data, window)
}

#[tauri::command]
pub fn update_fee_and_services(data: FeeAndServices, window: tauri::Window) -> Result<(), String> {
    fee_and_services::update_fee_and_services_db(data, window)
}

#[tauri::command]
pub fn get_fee_and_services(window: tauri::Window) -> Result<FeeAndServices, String> {
    fee_and_services::get_fee_and_services_db(window)
}
